.github/workflows/build-upstream-nightly.yml

```diff
@@ -17,3 +17,26 @@ jobs:
     secrets:
       SERVER_SSH_KEY: ${{ secrets.SERVER_SSH_KEY }}
       REMOTE_USER: ${{ secrets.REMOTE_USER }}
+  compute-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.supported-version.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: graycoreio/github-actions-magento2/supported-version@allow_using_next_version
+        with:
+          kind: custom
+          custom_versions: "magento/project-community-edition:next"
+        id: supported-version
+  integration-check:
+    needs: [deploy, compute-matrix]
+    if: (inputs.repo || '') == 'https://upstream-nightly.mage-os.org'
```

Comment:
Isn't this always true?

Reply:
I think the job would also run for mirror builds... I was following the example of https://github.com/mage-os/generate-mirror-repo-js/blob/main/.github/workflows/build-preview-mirror.yml#L89-L90